app/uk/gov/hmrc/personaldetailsvalidation/AssociationMongoRepository.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.personaldetailsvalidation
+
+import com.google.inject.ImplementedBy
+
+import org.mongodb.scala.bson.conversions.Bson
+import org.mongodb.scala.model.Indexes.ascending
+import org.mongodb.scala.model.{Filters, FindOneAndUpdateOptions, IndexModel, IndexOptions, Updates}
+import org.mongodb.scala.model.Updates.combine
+import play.api.Logging
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.play.json.PlayMongoRepository
+import uk.gov.hmrc.personaldetailsvalidation.formats.AssociationFormat.associationFormat
+import uk.gov.hmrc.personaldetailsvalidation.model.Association
+
+import java.util.concurrent.TimeUnit
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+trait AssociationRepositoryConstants {
+  val credentialId: String = "credentialId"
+  val sessionId : String = "sessionId"
+  val validationId: String = "validationId"
+  val lastUpdated: String = "lastUpdated"
+}
+
+@ImplementedBy(classOf[AssociationMongoRepository])
+trait AssociationRepository extends AssociationRepositoryConstants {
+  def create(association: Association): Future[Unit]
+  def find(credentialId: String, sessionId: String): Future[Option[Association]]
+}
+
+@Singleton
+class AssociationMongoRepository @Inject() (config: PersonalDetailsValidationMongoRepositoryConfig,
+                                            mongoComponent: MongoComponent)(implicit executionContext: ExecutionContext)
+  extends PlayMongoRepository[Association](
+    collectionName = "association",
+    mongoComponent = mongoComponent,
+    domainFormat = associationFormat,
+    indexes = Seq(
+      IndexModel(
+        ascending("lastUpdated"),
+        indexOptions = IndexOptions().name("expireAfterSeconds").expireAfter(config.collectionTtl.getSeconds, TimeUnit.SECONDS)
```

Comment:
this is quite important that it remains this config value, can we assert this in some way